apps/spruce/src/pages/image/ToolchainsTable/ToolchainsTable.test.tsx

```diff
@@ -0,0 +1,276 @@
+import { MockedProvider } from "@apollo/client/testing";
+import { RenderFakeToastContext } from "context/toast/__mocks__";
+import {
+  ImageToolchainsQuery,
+  ImageToolchainsQueryVariables,
+} from "gql/generated/types";
+import { IMAGE_TOOLCHAINS } from "gql/queries";
+import {
+  renderWithRouterMatch as render,
+  screen,
+  userEvent,
+  waitFor,
+  within,
+} from "test_utils";
+import { ApolloMock } from "types/gql";
+import { ToolchainsTable } from ".";
+
+const wrapper = ({ children }: { children: React.ReactNode }) => (
+  <MockedProvider
+    mocks={[
+      imageToolchainsMock,
+      imageToolchainsNextPageMock,
+      imageToolchainsFilterMock,
+    ]}
+  >
+    {children}
+  </MockedProvider>
+);
+
+describe("toolchains table", () => {
+  it("shows name field data", async () => {
+    const { Component } = RenderFakeToastContext(
+      <ToolchainsTable imageId="ubuntu2204" />,
+    );
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("toolchains-table-row")).toHaveLength(10);
+    });
+    const expectedNames = (
+      imageToolchainsMock.result?.data?.image?.toolchains.data || []
+    ).map(({ name }) => name);
+
+    const rows = screen.getAllByDataCy("toolchains-table-row");
+    expectedNames.forEach((expectedName, i) => {
+      expect(within(rows[i]).getAllByRole("cell")[0]).toHaveTextContent(
+        expectedName,
+      );
+    });
+  });
+
+  it("shows path field data", async () => {
+    const { Component } = RenderFakeToastContext(
+      <ToolchainsTable imageId="ubuntu2204" />,
+    );
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("toolchains-table-row")).toHaveLength(10);
+    });
+    const expectedPaths = (
+      imageToolchainsMock.result?.data?.image?.toolchains.data || []
+    ).map(({ path }) => path);
+
+    const rows = screen.getAllByDataCy("toolchains-table-row");
+    expectedPaths.forEach((expectedPath, i) => {
+      expect(within(rows[i]).getAllByRole("cell")[1]).toHaveTextContent(
+        expectedPath,
+      );
+    });
+  });
+
+  it("shows version field data", async () => {
+    const { Component } = RenderFakeToastContext(
+      <ToolchainsTable imageId="ubuntu2204" />,
+    );
+    render(<Component />, { wrapper });
+    await waitFor(() => {
+      expect(screen.queryAllByDataCy("toolchains-table-row")).toHaveLength(10);
+    });
+    const expectedVersions = (
+      imageToolchainsMock.result?.data?.image?.toolchains.data || []
+    ).map(({ version }) => version);
+
+    const rows = screen.getAllByDataCy("toolchains-table-row");
+    expectedVersions.forEach((expectedVersion, i) => {
+      expect(within(rows[i]).getAllByRole("cell")[2]).toHaveTextContent(
+        expectedVersion,
+      );
+    });
+  });
+
+  it("supports name filter", async () => {
```

Comment:
Just a question and not requesting a change: why did you make this test in unit tests instead of cypress?

Reply:
there is a test in Cypress, but since we can't predict how the image will change (e.g. the toolchain we try to search for in the Cypress tests could get completely deleted one day) we have to keep the Cypress test really general and search for a fake item. 

The unit test just makes sure that the parameters get passed properly to the GraphQL query